generation/src/generation.rs

```diff
@@ -122,27 +140,118 @@ impl Register {
 
         let cfg_attributes = &self.cfg_attributes;
 
-        syn::parse_quote! {
-            #doc_attribute
-            #(#cfg_attributes)*
-            pub fn #snake_case_name(&mut self) -> device_driver::RegisterOperation<'_, Self, #pascal_case_name, { #pascal_case_name::SIZE_BYTES }> {
-                device_driver::RegisterOperation::new(self)
+        match &resolved_register.kind {
+            RegisterKind::Register { .. } => {
+                let pascal_case_name =
+                    syn::Ident::new(&self.name.to_case(Case::Pascal), Span::call_site());
+                Ok(syn::parse_quote! {
+                    #doc_attribute
+                    #(#cfg_attributes)*
+                    pub fn #snake_case_name(&mut self) -> device_driver::RegisterOperation<'_, Self, #pascal_case_name, { #pascal_case_name::SIZE_BYTES }> {
+                        device_driver::RegisterOperation::new(self)
+                    }
+                })
+            }
+            RegisterKind::Block {
+                base_address,
+                repeat,
+                ..
+            } => {
+                let mod_name = syn::Ident::new(
+                    &resolved_register.name.to_case(Case::Snake),
+                    Span::call_site(),
+                );
+                let block_name = syn::Ident::new(
+                    &resolved_register.name.to_case(Case::Pascal),
+                    Span::call_site(),
+                );
+                let base_address = proc_macro2::Literal::u64_unsuffixed(base_address.unwrap_or(0));
+                if let Some(repeat) = repeat {
+                    let count = proc_macro2::Literal::u64_unsuffixed(repeat.count);
+                    let stride = proc_macro2::Literal::u64_unsuffixed(repeat.stride);
+
+                    Ok(syn::parse_quote! {
+                        #doc_attribute
+                        #(#cfg_attributes)*
+                        pub fn #snake_case_name(&mut self, i: <Self as AddressableDevice>::AddressType) -> #mod_name::#block_name<'_, Self> {
+                            assert!(i < #count);
+                            #mod_name::#block_name::new(self, #base_address + i * #stride)
+                        }
```

Comment:
I'm unsure whether it's better to use a usize as an argument or leave it like this.
It's sort of an indexing operation, which is normally done with usize.
But this way there's no casting...

Reply:
Yeah, I think the address type is correct here. `usize` is for indexing into memory, whereas this is indexing into the register space.

Reply:
I guess usize is correct variable width integer for indexing into computer memory. Adress type is also variable width, but for the registers.

I see your point. Let's leave it as is